functable.c

```diff
@@ -8,450 +8,497 @@
 #include "crc32_braid_p.h"
 #include "deflate.h"
 #include "deflate_p.h"
-
 #include "functable.h"
-
 #include "cpu_features.h"
 
-Z_INTERNAL Z_TLS struct functable_s functable;
```

Comment:
Where did global `struct functable_s functable` go?

Reply:
It was always at the end of the file. I find it absolutely strange that this code compiles in c, in c++ that's a compilation error.

Reply:
Redefining something to same type is silently ignored as forward declaring in C. This allows resolving possible circular dependency.